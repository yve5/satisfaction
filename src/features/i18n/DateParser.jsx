import DateParser from './components/DateParser';
export default DateParser;
