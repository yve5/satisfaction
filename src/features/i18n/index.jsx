import I18n from './components/I18n';

export default I18n;
