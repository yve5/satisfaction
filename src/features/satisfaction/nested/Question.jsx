import React from 'react';
import { connect } from 'react-redux';

import { changeValue } from '../actions/All';
import { SATISFACTION_RANGE_STEP } from '../resources/constants';

class Question extends React.Component {
  state = {
    value: 0,
  };

  componentDidMount() {
    const { id, changeValueProp } = this.props;
    changeValueProp(id, this.state.value);
  }

  render() {
    const { id, label, changeValueProp } = this.props;

    return (
      <div className="row mb-5 mb-md-3">
        <label className="col-9 col-md-7 form-label" htmlFor={`question-${id}`}>
          {label}
        </label>
        <div className="col-3 col-md-1 text-right">{this.state.value}%</div>
        <div className="col-md-4">
          <input
            className="form-range px-0 w-100"
            id={`question-${id}`}
            onChange={({ target: { value } }) => {
              changeValueProp(id, value);
              this.setState({ value });
            }}
            step={SATISFACTION_RANGE_STEP}
            type="range"
            value={this.state.value}
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  root: state,
});

const mapDispatchToProps = (dispatch) => ({
  changeValueProp: (id, value) => dispatch(changeValue(id, value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Question);
