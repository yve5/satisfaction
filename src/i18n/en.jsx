const I18N_DICTIONARY = {
  LANG: 'en',
  'Are you happy with your salary?': 'Are you happy with your salary?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'Are you satisfied with the interaction with people inside or outside the company?',
  'Confidence factor': 'Confidence factor',
  'Confidence factor calculation': 'Confidence factor calculation',
  'Do you actually enjoy doing the tasks of your job?':
    'Do you actually enjoy doing the tasks of your job?',
  'Do you appreciate the quality of the teamwork?':
    'Do you appreciate the quality of the teamwork?',
  'Do you have time for your personal projects?':
    'Do you have time for your personal projects?',
  'Do you like your workplace?': 'Do you like your workplace?',
  'Do your career prospects suit you?': 'Do your career prospects suit you?',
  'Do your colleagues agree with you?': 'Do your colleagues agree with you?',
  'Does doing this work bring you closer to your career goals?':
    'Does doing this work bring you closer to your career goals?',
  'Is your company doing well?': 'Is your company doing well?',
  Print: 'Print',
  Result: 'Result',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'The job is for you.',
  'Think twice about taking the job, or leaving it right away.':
    'Think twice about taking the job, or leaving it right away.',
  'You got the perfect job!': 'You got the perfect job!',
  "You're on the right track to finding the perfect job.":
    "You're on the right track to finding the perfect job.",
};

export default I18N_DICTIONARY;
