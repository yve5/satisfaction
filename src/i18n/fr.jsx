const I18N_DICTIONARY = {
  LANG: 'fr',
  'Are you happy with your salary?': 'Votre salaire vous convient-il ?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'L’intéraction avec les personnes en ou hors de l’entreprise vous satisfait-elle ?',
  'Confidence factor': 'Facteur de confiance',
  'Confidence factor calculation': 'Calcul du facteur de confiance',
  'Do you actually enjoy doing the tasks of your job?':
    'Aimez-vous réellement faire les tâches de votre poste ?',
  'Do you appreciate the quality of the teamwork?':
    'Appréciez-vous la qualité du travail d’équipe ?',
  'Do you have time for your personal projects?':
    'Avez-vous du temps pour vos projets personnels ?',
  'Do you like your workplace?': 'Aimez-vous votre lieu de travail ?',
  'Do your career prospects suit you?':
    'Vos perspectives d’évolution professionnelles vous convient-elles ?',
  'Do your colleagues agree with you?':
    'Vos collègues sont-ils d’accord avec vous ?',
  'Does doing this work bring you closer to your career goals?':
    'Exercer ce travail vous rapproche-t-il de vos objectifs professionnels ?',
  'Is your company doing well?': 'Votre société se porte-t-elle bien ?',
  lang: 'fr',
  languageId: 'Français',
  Print: 'Impression',
  Result: 'Résultat',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'Le job vous correspond.',
  'Think twice about taking the job, or leaving it right away.':
    'Songez à deux fois avant d’accepter le job, ou le quitter sur-le-champ.',
  'You got the perfect job!': 'Vous avez le job parfait !',
  "You're on the right track to finding the perfect job.":
    'Vous êtes sur la bonne voie pour trouver le job parfait.',
};

export default I18N_DICTIONARY;
