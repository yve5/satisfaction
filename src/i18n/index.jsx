import testingReducer from './testingReducer';

export default testingReducer;
