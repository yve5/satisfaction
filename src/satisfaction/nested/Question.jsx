import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { I18n } from 'organe/i18n';

import * as All from '../actions/All';
import { SATISFACTION_RANGE_STEP } from '../resources/constants';

const Question = ({ id, label, changeValue }) => {
  const [answer, setAnswer] = useState(0);

  useEffect(() => {
    changeValue(id, answer);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="row mb-5 mb-md-3">
      <label className="col-9 col-md-7 form-label" htmlFor={`question-${id}`}>
        <I18n>{label}</I18n>
      </label>
      <div className="col-3 col-md-1 text-right">{answer}%</div>
      <div className="col-md-4">
        <input
          type="range"
          value={answer}
          id={`question-${id}`}
          step={SATISFACTION_RANGE_STEP}
          className="form-range px-0 w-100"
          onChange={({ target: { value } }) => {
            setAnswer(value);
            changeValue(id, value);
          }}
        />
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  root: state,
});

const mapDispatchToProps = {
  ...All,
};

export default connect(mapStateToProps, mapDispatchToProps)(Question);
