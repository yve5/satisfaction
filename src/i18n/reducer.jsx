import { i18nReducer } from 'lapwing';
import fr from './fr';
import en from './en';

export default i18nReducer([en, fr]);
