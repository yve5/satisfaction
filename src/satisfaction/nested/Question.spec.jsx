import React from 'react';
import configureMockStore from 'redux-mock-store';

import { Provider } from 'react-redux';
import { act, fireEvent, render } from '@testing-library/react';

import i18n from '../../i18n';
import Question from './Question';

const mockStore = configureMockStore();

describe('Question', () => {
  it('should match expected snapshot. default', async () => {
    const { asFragment, getByTestId } = render(
      <Provider store={mockStore({ i18n })}>
        <Question
          id="culture"
          label="Do you have time for your personal projects?"
        />
      </Provider>
    );

    await act(async () => {
      fireEvent.change(getByTestId('s10n-question-range'), {
        target: { value: '75' },
      });
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
