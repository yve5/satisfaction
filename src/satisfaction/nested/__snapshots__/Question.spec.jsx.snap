// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match expected snapshot. default 1`] = `
<DocumentFragment>
  <div
    class="row mb-5 mb-md-3"
  >
    <label
      class="col-9 col-md-7 form-label"
      for="question-culture"
    >
      Avez-vous du temps pour vos projets personnels ?
    </label>
    <div
      class="col-3 col-md-1 text-right"
    >
      75%
    </div>
    <div
      class="col-md-4"
    >
      <input
        class="form-range px-0 w-100"
        data-testid="s10n-question-range"
        id="question-culture"
        step="1"
        type="range"
        value="75"
      />
    </div>
  </div>
</DocumentFragment>
`;
